src/Common/PODArray.h

```diff
@@ -21,6 +21,20 @@
 
 #include <Common/PODArray_fwd.h>
 
+template <typename T1, typename T2>
+constexpr bool allow_memcpy = std::is_same_v<T1, T2>;
```

Comment:
The name "allow_memcpy" is not good for DB namespace.
See: https://github.com/ClickHouse/ClickHouse/pull/12278/commits/6f8a274ba04fdec65f168ebe63f1acd715bdbd72

Yes, we should merge this PR, it makes sense.